src/components/Synthetics/TradeBox/hooks/useTradeboxTransactions.tsx

```diff
@@ -138,9 +217,45 @@ export function useTradeboxTransactions({ setPendingTxns }: TradeboxTransactions
         typeof allowedSlippage !== "number"
       ) {
         helperToast.error(t`Error submitting order`);
+        metrics.sendMetric({
+          event: `${metricType}.fail`,
+          isError: true,
+          message: "Error submitting order",
+        });
         return Promise.resolve();
       }
 
+      const metricData = {
+        metricType,
+        marketAddress: marketInfo?.marketTokenAddress,
+        initialCollateralTokenAddress: fromToken?.address,
+        initialCollateralDeltaAmount: increaseAmounts?.initialCollateralAmount,
+        targetCollateralAddress: collateralToken?.address,
+        collateralDeltaAmount: increaseAmounts?.collateralDeltaAmount,
+        swapPath: increaseAmounts?.swapPathStats?.swapPath || [],
+        sizeDeltaUsd: increaseAmounts?.sizeDeltaUsd,
+        sizeDeltaInTokens: increaseAmounts?.sizeDeltaInTokens,
+        triggerPrice: isLimit ? triggerPrice : 0n,
+        acceptablePrice: increaseAmounts?.acceptablePrice,
+        isLong,
+        orderType,
+        executionFee: executionFee?.feeTokenAmount,
+      };
```

Comment:
maybe create some metricData contructor function based on metric type?

right on it seems quite random. Also metricId could be contracted in same function. Returned type could depend on metric type with functino overloading, or at least union type